UANodeManager.cs

```diff
@@ -543,26 +697,18 @@ private ServiceResult DeleteAsset(ISystemContext context, MethodState method, IL
                 return new ServiceResult(StatusCodes.BadInvalidArgument);
             }
 
+            var isDeleted = false;
             IEnumerable<string> WoTFiles = Directory.EnumerateFiles(Path.Combine(Directory.GetCurrentDirectory(), "settings"), "*.jsonld");
             foreach (string file in WoTFiles)
             {
                 try
                 {
-                    string contents = File.ReadAllText(file);
-
-                    // check file type (WoT TD or DTDL)
-                    if (contents.Contains("\"@context\": \"dtmi:dtdl:context;2\""))

```

Comment:
Please keep the DTDL support.

Reply:
I think DTDL support isn't needed here since the file is deleted by the assetId in the file name and it doesn't care about the contents at all.

Reply:
I think, there are multiple strategies possible to check if there is a (valid) TD. 

1. File extension:  *.td.json, *.td.jsonld
2. content type: [application/td+json](https://www.iana.org/assignments/media-types/application/td+json)
3. @context check: https://www.w3.org/2022/wot/td/v1.1 is present
4. JSON Schema check: apply official [TD JSON Schema](https://www.w3.org/TR/wot-thing-description11/#json-schema-for-validation)  

Option 4 would be the best solution to be on the safe side in terms of trouble-free parsing.

Reply:
As the ConfigureAsset method always creates a .jsonld file in the settings folder, I'd say we can safely execute the DeleteAsset method by only referencing the assetId. This means that any kind of checking before removal is unnecessary as long as the assetId matches what should be removed from the settings folder, no?

Reply:
> As the ConfigureAsset method always creates a .jsonld file in the settings folder, I'd say we can safely execute the DeleteAsset method by only referencing the assetId. This means that any kind of checking before removal is unnecessary as long as the assetId matches what should be removed from the settings folder, no?

I think Sebastian's point was to make the validation better than it currently is, that's all...

Reply:
> > I think DTDL support isn't needed here since the file is deleted by the assetId in the file name and it doesn't care about the contents at all.
> 
> Yes, but we want to keep this here as a code example.

Of course I can leave/add DTDL support there (in the DeleteAsset method), but what would it do in this case?
I can't really find any useful example on what to compare the input argument (assetId) to inside the parsed object unless we further change something. Ideas?

Reply:
Please simply compare the DTDL filename against the asset id and make sure the asset id ends up in the file name.